### PR TITLE
Updating Bor Bor, making it math nerd

### PR DESCRIPTION
**Pull request by Sindorman**
This PR does these things:

- Adds new command `pick` which picks between random choice between n number of choices separated by commas.

- Adds new command `math` which use all of python in-built math functions and `math` library functions too. Uses python syntax as input

- Adds new command `math_functions` which gives list of all functions it can run and examples of how to use `math`

- Adds `requirements-install.bat` script that is used to install requirements automatically, given that user has Python 3.6+ installed with pip3.

- memes command now displays skull and dagger emojies if user types `!memes skull`

### CODE REVIEW

File: cogs/silly.py [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
 import aiohttp
 import discord
+from math import *
+import re
+import operator
 from discord.ext import commands
 
 from .utils import auth, AuthPerms, AuthType
 
+safe_dict = dict()
```
Comment (skull132) [inline cogs/silly.py]:
```suggestion
safe_dict = {}
```

(This is faster, believe it or not.)

File: cogs/silly.py [hunk 2]
```diff
@@ -159,6 +180,40 @@ def __init__(self, bot):
...
+
+    @commands.command()
+    @commands.cooldown(1, 1, commands.BucketType.channel)
+    async def math_functions(self, ctx):
+        """List of all math functions for math command. Call this command to see the list."""
+        wrong_l = set(["+", "-", "*", "/", "^"])
+        func = set(safe_dict.keys()) - wrong_l
+        rand_function = random.choice(list(func))
+        clean = re.sub('[\'{}]', '', str(func))
+        await ctx.send("```List of all math functions:\n{}\n\nUse them like {}(number or function)\nEach function needs brackets\nOr you can use operators like (2 + 5 / 4) ^ 2```".format(clean, rand_function))
```
Comment (skull132) [inline cogs/silly.py]: f-string this.

File: cogs/silly.py [hunk 3]
```diff
@@ -159,6 +180,40 @@ def __init__(self, bot):
...
+        if len(choices) == 0:
+            await ctx.send("Give me choices to pick from!")
+            return
+        lucky = random.choice(choices)
+        await ctx.send(lucky)
+
+    @commands.command()
+    @commands.cooldown(1, 5, commands.BucketType.channel)
+    async def math(self, ctx, *, inp: str):
+        """Does math. Uses python syntax with math library. Use like \"math 5 + 3\" or \"math [function](5)\". To get list of all functions use math_functions command."""
```
Comment (skull132) [inline cogs/silly.py]: Are these escaped quotes needed?

**Review by skull132 — CHANGES_REQUESTED** [review]
Move the maths stuff into its own cog. (Either a maths cog, or a more generic utilities cog.) Remove the globals and global init -> move it into the new cog's init function.

**Comment by Sindorman** [comment]
Issues addressed

File: cogs/math.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+class MathCog:
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    @commands.cooldown(1, 5, commands.BucketType.channel)
+    async def math(self, ctx, *, inp: str):
+        """Does math. Uses python syntax with math library. Use like math 5 + 3 * 2 or math [function](5). To get list of all functions use math_functions command."""
+        if inp == "":
+            await ctx.send(f"You need to give some numbers and functions to me")
```
Comment (skull132) [inline cogs/math.py]:
```suggestion
            await ctx.send("You need to give some numbers and functions to me")
```
not an f-string.

**Review by skull132 — APPROVED** [review]

File: cogs/math.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        clean = re.sub('[\'{}]', '', str(func))
+        await ctx.send(f"```List of all math functions:\n{clean}\n\nUse them like {rand_function}(number or function)\nEach function needs brackets\nOr you can use operators like (2 + 5 / 4) ^ 2```")
+
+class NumericStringParser(object):
+    '''
+    Most of this code comes from the fourFn.py pyparsing example
+
+    '''
+
+    def pushFirst(self, strg, loc, toks):
```
Comment (skull132) [inline cogs/math.py]: Please use snake_case for function names.

File: cogs/math.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        self.exprStack.append(toks[0])
+
+    def pushUMinus(self, strg, loc, toks):
```
Comment (skull132) [inline cogs/math.py]: Please use snake_case for function names.

File: cogs/math.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                   "log": math.log,
+                   "ceil": math.ceil,
+                   "floor": math.floor,
+                   "sqrt": math.sqrt,
+                   "abs": abs,
+                   "trunc": lambda a: int(a),
+                   "round": round,
+                   "sgn": lambda a: abs(a) > epsilon and cmp(a, 0) or 0}
+
+    def evaluateStack(self, s):
```
Comment (skull132) [inline cogs/math.py]: snake_case

File: cogs/math.py [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+        op = s.pop()
+        if op == 'unary -':
+            return -self.evaluateStack(s)
+        if op in "+-*/^%":
```
Comment (skull132) [inline cogs/math.py:126]:
```suggestion
        elif op in "+-*/^%":
```

File: cogs/math.py [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+        elif op == "E":
+            return math.e  # 2.718281828
+        elif op in self.fn:
+            return self.fn[op](self.evaluateStack(s))
+        elif op[0].isalpha():
+            return 0
+        else:
+            return float(op)
+
+    def eval(self, num_string, parseAll=True):
```
Comment (skull132) [inline cogs/math.py]:
```suggestion
    def eval(self, num_string, parse_all=True):
```

File: cogs/math.py [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+        self.exprStack = []
```
Comment (skull132) [inline cogs/math.py]:
```suggestion
        self.expr_stack = []
```

**Review by skull132 — CHANGES_REQUESTED** [review]
Snake case for var and method names please and thank you. PascalCase for class names is fine.

**Review by skull132 — APPROVED** [review]

**Pull request merged**